docs/docs/vulnerability/languages/java.md

```diff
@@ -0,0 +1,51 @@
+# Java
+
+## Features
+Trivy supports three types of Java scanning: `jar/war/par/ear`, `pom.xml` and `*gradle.lockfile` files.
+The following table provides an outline of the features Trivy offers.
+
+
+| Artifact         |  Maven repository [^1]   | Dev dependencies |
+|------------------|:------------------------:|:-----------------|
+| jar/war/par/ear  |            ✓             | Include          |
+| pom.xml          |            ✓             | Exclude          |
+| *gradle.lockfile |            -             | Exclude          |
+
+### Jar
+To find information about your jar[^2] file, Trivy parses `pom.properties` and `MANIFEST.MF` files in your jar[^2] file and takes `required properties`[^3].
+
+If those files don't exist or don't contain enough information - Trivy uses [maven repository REST API](https://central.sonatype.org/search/rest-api-guide) to try to get missing information.
+
+!!! Note
+    Trivy only takes information about package. We don't take list of vulnerability for package from `maven repository`(Information about data sources for java you can see [here](../detection/data-source.md))
+
+
+You can disable use of `maven repository REST API`. Use `--offline-scan` flag for this. `--offline-scan` flag doesn't affect Trivy-DB.
+
+!!! Warning
+    Trivy may skip some dependencies(that were not found in `pom.properties` or `MANIFEST.MF`) when the `--offline-scan` flag is used.
+
+Base jar[^2] may contain inner jars[^2] within itself. To find information aobut these jars[^2] same logic is used as for base jar[^2].
+
+### Pom.xml
+Trivy parses your `pom.xml` file and try to find files with dependencies on your local PC(project folder[^4], relativePath field[^5] and local repository folder[^6]).
+
+If your local PC doesn't have necessary files - Trivy try to find info about these dependencies in [maven repository](https://repo.maven.apache.org/maven2/).
+
+!!! Note
```

Comment:
I think this is the same comment as earlier -- why does it need to be repeated?

Reply:
Some users only work with `pom.xml` files.
And it is possible that user will  check only the `pom.xml` part of document and not check `jar` part.
Then he will miss this important information or will think that this information is used only for `jars`.